src/v/rpc/test/rpc_gen_cycling_test.cc

```diff
@@ -363,6 +364,35 @@ FIXTURE_TEST(timeout_test, rpc_integration_fixture) {
     client.stop().get();
 }
 
+FIXTURE_TEST(timeout_test_cleanup_resources, rpc_integration_fixture) {
+    configure_server();
+    register_services();
+    start_server();
+
+    rpc::client<echo::echo_client_protocol> client(client_config());
+    client.connect(model::no_timeout).get();
+    using units_t = std::vector<ssx::semaphore_units>;
+    ::mutex lock;
+    units_t units;
+    units.push_back(std::move(lock.get_units().get()));
+
+    auto opts = rpc::client_opts(rpc::clock_type::now() + 1s);
+    opts.resource_units = ss::make_foreign(
+      ss::make_lw_shared(std::move(units)));
+
+    // Resources should now be owned by the client.
+    BOOST_REQUIRE(lock.try_get_units().has_value() == false);
+    auto echo_resp = client.sleep_for(
+      echo::sleep_req{.secs = 10}, std::move(opts));
+    BOOST_REQUIRE_EQUAL(
+      echo_resp.get0().error(), rpc::errc::client_request_timeout);
+    // Verify that the resource is released before the RPC finishes.
+    tests::cooperative_spin_wait_with_timeout(5s, [&] {
+        return lock.try_get_units().has_value();
+    }).get();
```

Comment:
how does this verify that the resource is related before the RPC finishes? `echo_resp.get0()` blocks until a response is received, and only after that, do we do a test of the lock.

Reply:
you're right, that comment is incorrect.. I'll fix it (just to be clear.. the intent was to add coverage for release of units after RPC finishes with a timeout.).